### ISSUE TITLE
After update from 2.3.7 to 2.4.0, on some note played via tuxguitar, a "larsen" is emitted shortly

### ISSUE THREAD

**Issue by ycollet** [FluidSynth/fluidsynth#1451]
### FluidSynth version
I used fuildsynth 2.4.0 from Fedora 41. The version before this update was 2.3.7.

### Describe the bug
I used fluidsynth via tuxguitar.
When I play some guitarpro score, some some notes; a larsen is emitted briefly.
Seems to disappear if I turn off chorus and / or reverb.

### Expected behavior
No audible larsen

### Steps to reproduce
I haven't yet a good reproducer. I will pos one once I found a reproductible score

### Additional context
There was no problems before 2.4.0. Tuxguitar was not updated in the mean time.
I removed the tuxguitar configuration and start a new one from scratch but the problem is still here.

**Comment by derselbst** [comment]
What do you mean by "larsen"? Also, I highly recommend to use fluidsynth 2.4.1 since we have fixed possible clicks and pops in the low pass filter. If it also happens with 2.4.1, pls. post at least an audio sample where the "larsen" can be heard, or even better, a reproducible score.

**Comment by ycollet** [comment]
I will wait for the fedora update.
From tiime to time, there is a high frequency sound really similar to a larsen we can hear with a guitar.
The problem is not always reproductible. Once I found a GuitarPro score where the problem occurs regularly, I will record the sound.

**Comment by ycollet** [comment]
Here is an example.

![image](https://github.com/user-attachments/assets/98063fdb-e390-4c97-9435-967641d28d98)

We can here the high frequency sound in the 2 first sequences.

https://github.com/user-attachments/assets/673b39db-4a1f-46db-8ff1-eda82fa1aac1

**Comment by derselbst** [comment]
So, you mean that feedback that builds up on the right channel? To me, this
sounds like a note that hasn't received a noteOff yet. I need a score or a
MIDI file to investigate. With which version of fluidsynth did you produce
this? Also, it would be helpful if you posted an example of what you would
expect to sound it like.

**Comment by mrbumpy409** [comment]
It sounds to me like the MIDI file is using the GS preset **008:030 Feedback Guitar**, which has a feedback that creeps in as the notes sustain.

**Comment by ycollet** [comment]
Good idea. I will check tonight if it's the case.

**Comment by ycollet** [comment]
I reproduced the problem via an exported midi file played with vlc / fluidsynth and a specific sound font.
I uploaded the files on my website.
The soundfont: http://ycollette.free.fr/Guitare/63mg_The_Xioad_Bank.sf2
The midi file: http://ycollette.free.fr/Guitare/bolt_thrower_armageddon_bound_transposed.mid
When played on vlc + fluindsynth plugin, the problem starts at 3:40

**Comment by derselbst** [comment]
Ok, thanks. So I'm still trying to understand what you mean. For this, I've rendered the midi + SF2 with both, fluidsynth 2.3.7 and 2.4.2. Then I inverted one rendering, mixed them both together and exported it to this file, so we only hear the differences:
[1451.zip](https://github.com/user-attachments/files/18311654/1451.zip)

The differences in the drums are expected, due to #1199. And yes, fluidsynth 2.4.2 produces a strange sinusoidal feedback at e.g. 4:28 and 4:41 on the right channel - is that what you mean by "larsen"?

**Comment by ycollet** [comment]
Yes, by feedback I mean this sine wave on the right channel at the time you specified.

**Comment by derselbst** [comment]
Thanks for confirming I'll investigate what's going on. May take a while though...

**Comment by ycollet** [comment]
OK, thanks a lot for your work :)

**Comment by spessasus** [comment]
Since you've mentioned that the issue dissappears with the effects disabled, it seems that the only change reverb has received is:
![image](https://github.com/user-attachments/assets/b40ffcf7-79dc-4ca5-acf3-12c42b3169df)

**Comment by ycollet** [comment]
I will check again and keep you informed. I don't know if the vlc / fluidsynth plugin add chorus / reverb ?

**Comment by spessasus** [comment]
![image](https://github.com/user-attachments/assets/6b8df9a5-328b-4d59-9302-916adf63d72a)
You can enable or disable it in the preferences (make sure to select all preferences)

**Comment by derselbst** [comment]
The provided difference audio was generated with effects disabled, so it's not related to this.

**Comment by spessasus** [comment]
Splitting the file into two where:
one contains percussion and bass (channels 10, 12 and 13)
one contains the rest (distortion guitars)

results in a clean output. Adding them back together results in clear sound that 2.3.7 produces.

**Comment by spessasus** [comment]
attaching them for reference:
[Archive.zip](https://github.com/user-attachments/files/18382169/Archive.zip)

**Comment by derselbst** [comment]
I have identified 186a4af987d123cd59ca3c3abf1c1f0a6174427a (#1314) as the culprit. I still need to understand what's going on exactly, such as why the problem disappears if e.g. I only accept noteOn events from either channel 5 or channel 4 (which are the ones causing the feedback).

**Comment by ycollet** [comment]
Excellent !
A big thanks for this work !!

**Comment by derselbst** [comment]
So, it turns out that when a voice is in delay phase, its amplitude is not updated. Since the amplitude isn't cleared either, it may inherit an amplitude as well as an amplitude increment of a previous voice. If that previous amp increment was non-zero, nothing prevents the amplitude of the current in-delay-phase-voice from rising infinitely. A case that I indeed haven't considered. I will propose a fix, since I was reworking that code anyway.

**Comment by ycollet** [comment]
Thanks !
Is there any donation page ? Like a patreon account ?

**Comment by derselbst** [comment]
No, sry, I thought about this before but decided against a sponsorship. But thanks for the offer :)

**Comment by derselbst** [comment]
Fixed by #1470 and will be released with 2.4.3. Thanks again for the report!

**Comment by ycollet** [comment]
Thanks a lot for your work !
